### PR TITLE
Master fixes

### PR DESCRIPTION
**Pull request by sithhell**
Small fixes to master:
 - cosmetic addition of promise_lco<T>::keep_alive default move assignment
 - Fixing c&p errors for operator> in optional after constexpr 'fixes'

### CODE REVIEW

File: hpx/lcos/detail/promise_base.hpp [hunk 1]
```diff
@@ -316,6 +316,8 @@ namespace lcos {
                   : ptr_(o.ptr_.release(), &wrapping_deleter)
                 {}
 
+                keep_alive& operator=(keep_alive&& o) = default;
+
```
Comment (hkaiser) [inline hpx/lcos/detail/promise_base.hpp]: Should we add a default copy assignment operator as well?

Comment (sithhell) [reply hpx/lcos/detail/promise_base.hpp]: No. This thing is obviously not copyable

Comment (hkaiser) [reply hpx/lcos/detail/promise_base.hpp]: Hmm, that's not really obvious to me.  If the intention is to inhibit copying, add a `=delete` for the corresponding functions.

**Comment by sithhell** [comment]
It's already implicitly deleted due to the `unique_ptr` member. This is an
internal helper only anyways and not intended for public use.

**Comment by hkaiser** [comment]
Ok, you're right.

**Pull request merged**